### PR TITLE
Allows security groups to be referenced by name and filters

### PR DESCRIPTION
**Pull request by iamemilio**
**What this PR does / why we need it**: Lets you specify either ID, Name, or a set of Filters for each entry in securityGroups following the syntax established in networks

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # 234

### CODE REVIEW

File: pkg/apis/openstackproviderconfig/v1alpha1/types.go [hunk 1]
```diff
@@ -105,6 +128,51 @@ type RootVolume struct {
 	Size       int    `json:"diskSize,omitempty"`
 }
 
+// OpenstackProviderSpecTemp is a workaround for unmarshalling clusterv1 providerSpec
+// Todo(egarcia): come up with a better unmarshalling scheme
+type OpenstackProviderSpecTemp struct {
```
Comment (flaper87) [inline pkg/apis/openstackproviderconfig/v1alpha1/types.go]: TB, I'm not sure I understand the need for this struct. Happy to chat about it when you're online.

File: pkg/cloud/openstack/clients/machineservice.go [hunk 2]
```diff
@@ -275,6 +277,83 @@ func CreatePort(is *InstanceService, name string, net ServerNetwork, securityGro
...
+	}
+	for _, element := range list {
+		if element == name {
+			return true
+		}
+	}
+	return false
+}
+
+func GetSecurityGroups(is *InstanceService, sg_param []openstackconfigv1.SecurityGroupParam) ([]string, error) {
```
Comment (flaper87) [inline pkg/cloud/openstack/clients/machineservice.go]: You may want to take a look at what I did for the `subnets` filter in https://github.com/kubernetes-sigs/cluster-api-provider-openstack/commit/2ee134e451af7c42c81ddd72693a320bff9b285c

**Comment by iamemilio** [comment]
Flaper87 merged changes. Closing

**Comment by flaper87** [comment]
@iamemilio I've added a commit on this branch that should help cleaning up the `GetSecurityGroups` function. Mind including it in your PR? https://github.com/flaper87/cluster-api-provider-openstack/tree/sec-groups

**Pull request merged**